open-machine-learning-jupyter-book/deep-learning/LSTM.md

```diff
@@ -13,4 +13,635 @@ kernelspec:
   name: python3
 ---
 
-# LSTM (TBD)
+# Long-Short Term Memory
+
+As we have learned the normal RNNs on the previous lessons, today we are going to use long-short term memory (LSTM) to fix the problems that are induced from the RNN architecture.
+
+```{seealso}
+The vanilla LSTM is proposed in 2005, the paper is [Bidirectional LSTM Networks for Improved Phoneme Classification and Recognition](https://link.springer.com/chapter/10.1007/11550907_126), and after that, a lot of paper based on it appeared.
+```
+
+## Overview
+The essential problems of RNN are vanishing/exploding gradient problems.The gradient vanishing/exploding problem is a common issue in training deep neural networks. This problem occurs when the gradient of the loss function with respect to the weights of the network becomes very small or very large.
+There are several possibilities will bring on this kind of problem, for example:
+- a poor choice of hyper-parameters
+- a poor architecture
+- a bug in the code
+
+Luckily, LSTM can use a memory cell for modeling long-range dependencies and avoid vanishing/exploding gradient problems.
+
+First, let's see the architecture of LSTM cell.
+
+:::{figure-md} LSTM cell
+<img src="../../images/deep-learning/LSTM/LSTM_cell.png" width="90%" class="bg-white mb-1">
+
+Illustration of LSTM unit
+:::

```

Comment:
PLease always add the reference to the image source.

Reply:
Done. I have added the reference to the reference.bib.